kuma/users/tests/test_views.py

```diff
@@ -388,10 +390,119 @@ def test_post_submits_no_revisions_to_akismet_when_wrong_revisions_in_request(se
         eq_(mock_requests.call_count, 0)
 
-#    TODO: Phase III:
-#    def test_post_reverts_revisions(self):
-#    def test_post_deletes_new_pages(self):
-#
+    def test_post_deletes_new_page(self):
+        """POSTing to ban_user_and_cleanup url with a new document."""
+        # Create a new document and revisions as testuser
+        # Revisions will be reverted and then document will be deleted.
+        new_document = create_document(save=True)
+        new_revisions = self.create_revisions(
+            num=3,
+            document=new_document,
+            creator=self.testuser)
+
+        # Pass in all revisions, each should be reverted then the
+        # document will be deleted as well
+        data = {'revision-id': [rev.id for rev in new_revisions]}
+
+        self.client.login(username='admin', password='testpass')
+        resp = self.client.post(self.ban_testuser_url, data=data)
+        eq_(200, resp.status_code)
+
+        # Test that the document was deleted successfully
+        deleted_doc = Document.admin_objects.filter(pk=new_document.pk).first()
+        eq_(deleted_doc.deleted, True)
+
+    def test_post_reverts_page(self):
+        """POSTing to ban_user_and_cleanup url with revisions to a document."""
+        # Create a new document and first revision as an admin
+        # and spam revisions as testuser.
+        # Document should be reverted with a new revision by admin.
+        new_document = create_document(save=True)
+        self.create_revisions(num=1, document=new_document, creator=self.admin)
+        spam_revisions = self.create_revisions(
+            num=3,
+            document=new_document,
+            creator=self.testuser)
+
+        # Before we send in the spam,
+        # last spam_revisions[] should be the current revision
+        eq_(new_document.current_revision.id, spam_revisions[2].id)
+        # and testuser is the creator of this current revision
+        eq_(new_document.current_revision.creator, self.testuser)
+
+        # Pass in all spam revisions, each should be reverted then the
+        # document should return to the original revision
+        data = {'revision-id': [rev.id for rev in spam_revisions]}
+
+        self.client.login(username='admin', password='testpass')
+        resp = self.client.post(self.ban_testuser_url, data=data)
+        eq_(200, resp.status_code)
+
+        new_document = Document.objects.filter(id=new_document.id).first()
+        # Make sure that the current revision is not the spam revision
+        for revision in spam_revisions:
+            ok_(revision.id != new_document.current_revision.id)
+        # The most recent Revision object should be the document's current revision
+        latest_revision = Revision.objects.order_by('-id').first()
+        eq_(new_document.current_revision.id, latest_revision.id)
+        # Admin is the creator of this current revision
+        eq_(new_document.current_revision.creator, self.admin)
```

Comment:
Can you add a check that ``new_document.current_revision.content`` is the same as the original content?